runtime/jcl/common/jclcinit.c

```diff
@@ -644,6 +644,15 @@ initializeRequiredClasses(J9VMThread *vmThread, char* dllName)
 	}
 #endif /* defined(J9VM_OPT_OPENJDK_METHODHANDLE) */
 
+#if JAVA_SPEC_VERSION >= 19
+	/* These fields point to the next and previous VirtualThreads in the liveVirtualThreadList. */
+	if (0 != vmFuncs->addHiddenInstanceField(vm, "java/lang/VirtualThread", "linkNext", "Ljava/lang/VirtualThread;", &vm->virtualThreadLinkNextOffset)) {
+		return 1;
+	}
+	if (0 != vmFuncs->addHiddenInstanceField(vm, "java/lang/VirtualThread", "linkPrevious", "Ljava/lang/VirtualThread;", &vm->virtualThreadLinkPreviousOffset)) {
+		return 1;
+	}
```

Comment:
https://github.com/eclipse-openj9/openj9/blob/f98284648fb7e6832c7b8f38603ec4b748196da4/runtime/vm/resolvefield.cpp#L507-L509

For the hidden fields, `GC will not automatically know about it`: If the GC moves the objects stored in these hidden fields, then the object ref stored in the hidden field will become stale. @LinHu2016 Is it possible for the GC to recognize the objects stored in the hidden fields? If so, how?

Reply:
I though hidden fields are walked by GC using `GC_MixedObjectDeclarationOrderIterator` which in turn calls:
```
fullTraversalFieldOffsetsStartDo
    fieldOffsetsStartDo
        countAndCopyHiddenFields
```
maybe the comment should be updated?
@LinHu2016 to confirm

Reply:
If the field is declared as object, the GC will update it.

Reply:
As @gacholio mentioned the field need to be declared as O-slot(reference, also keep in mind it could be compressed ref, so you could use GC_SlotObject.readReferenceFromSlot()/writeReferenceToSlot()  to access it), then GC treat both links as the reference, mark/copy them,  update slots and also scan them.

Reply:
GC might not scan the above hidden fields, but must be updated always